### ISSUE TITLE
Tool.jar 1.3.9b update

### ISSUE THREAD

**Issue by owanneke** [mynttt/UpdateTool#27]
Since the 1.3.9b update, the docker container cannot run anymore.

2020-05-26 12:35:39 | stdout | [ERROR] - 2020-05-26 12:35:39 @ Main.lambda$main$1: The application will terminate now. (Version: 1.3.9b)
-- | -- | --
2020-05-26 12:35:39 | stdout | [ERROR] - 2020-05-26 12:35:39 @ Main.lambda$main$1: ========================================
2020-05-26 12:35:39 | stdout | at updatetool.Main.main(Main.java:76)
2020-05-26 12:35:39 | stdout | at updatetool.Main.processArgs(Main.java:135)
2020-05-26 12:35:39 | stdout | [ERROR] - 2020-05-26 12:35:39 @ Main.lambda$main$1: java.lang.IllegalArgumentException: Invalid arguments: [imdb-docker] \| Arguments must be an even number (currently: 1) (id + args for implementation) * called tools. Look at the help by running this tool without arguments for an example.
2020-05-26 12:35:39 | stdout | [ERROR] - 2020-05-26 12:35:39 @ Main.lambda$main$1: ========================================
2020-05-26 12:35:39 | stdout | [ERROR] - 2020-05-26 12:35:39 @ Main.lambda$main$1: Please contact the maintainer of the application with the stacktrace below if you think this is unwanted behavior.
2020-05-26 12:35:39 | stdout | [ERROR] - 2020-05-26 12:35:39 @ Main.lambda$main$1: Uncaught IllegalArgumentException exception encountered...
2020-05-26 12:35:39 | stdout | [INFO ] - 2020-05-26 12:35:39 @ Main.loadImplementation: Loaded implementation << imdb-docker << updatetool.imdb.ImdbDockerImplementation

**Comment by mynttt** [comment]
The CLI-Interface has changed to make this tool more future-proof to additions.

It appears that the docker container has downloaded the new jar while not being updated itself. Thus the entry point script still uses the old command and causes a startup failure.

Try updating your docker container, it should then have the new entry point that works.

Given this confusing error I might remove the auto-patching feature from the container startup and instead emit a warning that an older version is used.

**Comment by owanneke** [comment]
Hi,

deleting the container and reinstalling it did indeed fix the issue.
Although overwriting the container image would probably have been sufficient :-)

i'm not sure if the container should update part of its components; nor that logging a version conflict would be sufficient. 

I found out about the problem because the container crashes on run and the host generated a warning on that. Writing a conflict to the console/log would probably not be seen for some time.

imho the goal should be to keep the entire image in version sync?

**Comment by mynttt** [comment]
The patcher I added on startup was pretty naive, it did not patch the actual container but only the tool.jar file by modifying the containers local storage.

Which worked well until the CLI changed, since the entrypoint script was still calling the newly downloaded tool with the invalid argument format.

I removed the patching feature now to prevent something like that in the future. Only a warning will be emitted now. I think having the container patch itself is not a good decision as the error you had was not straightforward at all and will now be prevented in the future.
